### PR TITLE
Downgrade glibc to 2.24

### PR DESCRIPTION
**Pull request by chaitu236**
This patchset along with the one in `openembedded-core` downgrades glibc to 2.24 as glibc 2.33 has a locking bug on RT applications.

Azdo Bug: [1995395](https://ni.visualstudio.com/DevCentral/_workitems/edit/1995395)

### Testing

- Built locally and in a PR build
- Installed on a target
- Verified that bug doesn't reproduce

### CODE REVIEW

File: recipes-core/glibc/glibc_2.%.bbappend [hunk 1]
```diff
@@ -1,11 +1,17 @@
...
+
+SRC_URI =+ "file://0001-pi-condvars-add-protocol-support-to-pthread_condattr.patch \
+            file://0002-pi-condvars-add-tst-condpi1.c-basic-API-test.patch \
+            file://0003-pi-condvars-add-tst-condpi2.c-priority-inversion-avo.patch \
+            file://0004-benchtests-Add-benchmarks-for-pthread_cond_-function.patch \
+            file://0005-x86_64-Remove-assembly-implementations-for-pthread_c.patch \
+            file://0006-pi-condvars-EAGAIN-retry-in-pthread_cond_-broadcast-and-signal.patch \
+           "
 
 # Add patches to build the ja_JP.WINDOWS-31J and zh_CN.CP936 locales for
```
Comment (chaitu236) [inline recipes-core/glibc/glibc_2.%.bbappend:16]: I had to make slight modifications to @SparkingSpork 's patches so they apply on glibc-2.24 sources.

**Comment by amstewart** [comment]
@chaitu236 I've created a `nilrt/22.5/hardknott` release branch ref. Would you mind resubmitting this change against that ref?

**Review by amstewart — APPROVED** [review]

**Comment by chaitu236** [comment]
> @chaitu236 I've created a `nilrt/22.5/hardknott` release branch ref. Would you mind resubmitting this change against that ref?

Done

**Review by amstewart — APPROVED** [review]

File: recipes-core/glibc/glibc/cp936-gconv-modules.patch [hunk 2]
```diff
@@ -1,16 +1,6 @@
-From a33deff1b9fc055aa9cfc6b1569cb218742637d2 Mon Sep 17 00:00:00 2001
-From: Bill Pittman <bill.pittman@ni.com>
-Date: Fri, 19 Jun 2020 14:15:16 -0500
-
----
- iconvdata/gconv-modules | 9 ++++++---
- 1 file changed, 6 insertions(+), 3 deletions(-)
-
-diff --git a/iconvdata/gconv-modules b/iconvdata/gconv-modules
-index 16d09eb98d..a1279c8719 100644
```
Comment (SparkingSpork) [inline recipes-core/glibc/glibc/cp936-gconv-modules.patch:10]: Was removal of these lines intentional? I notice you do on some of the patches but not others.

**Review by SparkingSpork — APPROVED** [review]

File: recipes-core/glibc/glibc/cp936-gconv-modules.patch (hunk 2 above)
Comment (chaitu236) [reply recipes-core/glibc/glibc/cp936-gconv-modules.patch:10]: This was not intentional.. let me take a look.

Comment (chaitu236) [reply recipes-core/glibc/glibc/cp936-gconv-modules.patch:10]: These changes came directly from sumo version of these files which didn't have these comments.

**Comment by amstewart** [comment]
Merging this PR now, because it is a priority change.

**Pull request merged**